src/core/algorithms/fd/hyfd/hyfd.cpp

```diff
@@ -14,12 +14,16 @@
 #include "algorithms/fd/hycommon/util/pli_util.h"
 #include "inductor.h"
 #include "sampler.h"
+#include "thread_number/option.h"
 #include "validator.h"
 
 namespace algos::hyfd {
 
 HyFD::HyFD(std::optional<ColumnLayoutRelationDataManager> relation_manager)
-    : PliBasedFDAlgorithm({}, relation_manager) {}
+    : PliBasedFDAlgorithm({}, relation_manager) {
+    RegisterOption(config::kThreadNumberOpt(&threads_num_));
+    MakeOptionsAvailable({config::kThreadNumberOpt.GetName()});
```

Comment:
Logically, this is an execution stage option, so override `MakeExecuteOptsAvailable` and do this there.